teddy.js

```diff
@@ -552,791 +390,1250 @@
-                  elseCond = elseCond ? sibling + replaceNonRegex(elseCond[0], condString, '') + '</else>' : null
-                } else {
-                  findElses = false
-                  elseCond = false
-                }
+    // Return our rendered HTML file
+    return renderedTemplate
+  }
 
-                if (elseCond) {
-                  parts.push(elseCond)
-                  condString += elseCond
-                } else {
-                  findElses = false
-                }
-              } else {
-                findElses = false
-              }
-            }
-            while (findElses)
+  // Parse conditional teddy tags (i.e <if> <unless> <elseif> <elseunless> <else>)
+  function parseConditional (charList, type, model) {
+    var i // Template index
+    var j // Comment index
+    var l = charList.length // Length of template array
+
+    var currentChar // Current character in our template array
+    var tagStart = 0 // Index of '<' for the beginning of our conditional
+    var readingConditional = true // Start parsing contents of <if> <elseif> <unless> <elseunless> tags
+    var readMode = false // Start parsing literal equality condition (ex: <if something='here'>)
+    var outsideTags = false // Flag telling us we are inbetween condition tags
+    var isNested = false // Extra check for whether or not the nested condition also has an <else>
+    var nested = 0 // Keeps track of how many nested conditionals are present
+    var teddyVarName = '' // Teddy conditional argument name or operator used (i.e: or, and, xor, not)
+    var teddyVarExpected = '' // Literal value for a conditional teddy argument (i.e: <if something='some content'>)
+    var condition = {
+      type: type // <if> <unless>
+    }
 
-            result = renderConditional(condString, parts, model)
+    var boc = [] // Array of 2-length lists that contain the relevant indices for an open conditional tag [<, >]
+    var eoc = [] // Array of 2-length lists that contain the relevant indices for a closing conditionaltag [<, >]
+    var conditions = [] // List of objects containing all operators, variables and type of conditional (if or unless)
+    var varList = [] // List of objects containing relevant information about our conditions arguments
+    var operators = [] // List of operators used in a single conditional statement
+    var commentList = [] // Array of 2-length lists that contain the start/end indices for template comments inbetween conditionals
+    var commentIndices = [] // 2-length list containing start/end indices for template comments inbetween conditionals
+
+    var currentClosingTag = secondaryTags['celse' + type] // </elseif> </elseunless>
+    var currentOpenTag = primaryTags[type] // <if> <unless> <elseif> <elseunless>
+
+    // Look for begin/end of conditionals
+    for (i = currentOpenTag.length; i < l; i++) {
```

Comment:
Is there any speed up when caching lengths? Across all of instances in the rendering process?